opal/mca/btl/openib/btl_openib_mca.c

```diff
@@ -641,6 +641,9 @@ int btl_openib_register_mca_params(void)
         /* Don't try to recover from this */
         return OPAL_ERR_OUT_OF_RESOURCE;
     }
+    if (NULL != mca_btl_openib_component.default_recv_qps) {
+        free(mca_btl_openib_component.default_recv_qps);
+    }
     mca_btl_openib_component.default_recv_qps = default_qps;
```

Comment:
thanks @hjelmn
iirc, we do this (useless) test virtually everywhere else, so i did it again to keep some consistency in the code.

maybe we should all agree on a policy about that, for example
- keep testing vs ```NULL``` before ```free()```
- stop testing vs ```NULL``` before ```free()``` from now
- remove all tests vs ```NULL``` before ```free()``` (that could be quite a massive change)

@jsquyres @rhc54 @bosilca @hppritcha any thoughts ?